File: planning-idsdp-rms-frontend/src/app/main/modules/rpm/models/ResearcherProposalActionPlan.ts
export class ResearcherProposalActionPlan {
    id: number;
    //researcherProposal:any;
    proposalActionPlanId:number;   
    uuid: string;
    researcherProposalId: number;
    catWiseActPlanId:number;
    taskName: string;
    startDate: Date;
    endDate: Date;
    remarks: string;    
    isDeleted: 1 | 0;
    isAgree:true | false;
    totalDays:0;   
}

File: planning-idsdp-rms-rpm-backend/src/main/java/com/ibcs/idsdp/rpm/model/domain/ResearcherProposalActionPlan.java
package com.ibcs.idsdp.rpm.model.domain;

import java.time.LocalDate;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

import org.hibernate.annotations.Cache;
import org.hibernate.annotations.CacheConcurrencyStrategy;

import com.ibcs.idsdp.common.model.domain.BaseEntity;

import lombok.Data;

@Table(name = "m1_researcher_proposal_action_plan")
@Entity
@Data
@Cache(usage = CacheConcurrencyStrategy.READ_WRITE)
public class ResearcherProposalActionPlan extends BaseEntity {

	@NotNull
	@ManyToOne
	@JoinColumn(name = "researcher_proposal_id")
	private ResearcherProposal researcherProposal;

	@NotNull
	private String taskName;
	private Integer totalDays;
	private LocalDate startDate;
	private LocalDate endDate;
	private String remarks;
	private Boolean isAgree;
	@NotNull
	private Long catWiseActPlanId;
}
